### ISSUE TITLE
Request: d6 taking highest with 6s adding +1

### ISSUE THREAD

**Issue by TripleOakLeaf** [Rolisteam/DiceParser#88]
Is there a command or a way to make a dice roll that rolls a pool of dice taking the highest and when the top side is rolled on the dice more than once, each time adds +1....

Eg. 
5d6: 1,1,2,2,3 = 3
5d6: 2,3,4,4,4 = 4
5d6: 4,5,5,6,6 = 7
5d6: 1,6,6,6,6 = 9

You take the highest number but additional 6s add +1.

I'm trying to use it for Far Away Land and the Silhouette series of RPGs.

**Comment by SaltatorMortis** [comment]
jup its possible
here is the dice command:
`!5d6s;$1k1;$1c[$2]-1;$3+$2;"# $4 (@1)"`
in plain english 
roll 5d6, get the highest dice, count the amount of the highest dice and subtract 1

hope this helps, 
have fun playing

**Comment by SaltatorMortis** [comment]
ps
if you  want to use a alias you need to escape the inner `"` with `\`
that command would look like this:
`!alias add ROLL "5d6s;$1k1;$1c[$2]-1;$3+$2;\"# $4 (@1)\""`

**Comment by TripleOakLeaf** [comment]
Thank you so much. It indeed works. Quick question about the Alias part, Could the backend of the alias be used so I can change the input of the number of dice rolled each time? That would cover all the mechanics of the RPG.

eg. 
!alias add FAL "s;$1k1;$1c[$2]-1;$3+$2;\"# $4 (@1)\""
!5d6 FAL

or even better

!alias add F "d6s;$1k1;$1c[$2]-1;$3+$2;\"# $4 (@1)\""
!5F

I know it doesn't work but I'm not sure what change might make that work.

Regardless, this is awesome. A bunch of us have been hunting for a diceroller and I will pass the info around.

**Comment by SaltatorMortis** [comment]
ah sorry my misunderstanding-
as for the manual dicestack size you would need a macro command instead of alias

based on your template ive come up with this:
```!macro add Fal "d6s;$1k1;$1c[$2]-1;$2i:[6]{$3}{0};$2+(1*$4);\"# $5 (@1)\"" false
!5Fal```
or
```!macro add (\d+)F "d6s;$1k1;$1c[$2]-1;$2i:[6]{$3}{0};$2+(1*$4);\"# $5 (@1)\"" true
!5F
```
